faucet-server/Sources/App/Controllers/CKBController.swift

```diff
@@ -28,35 +28,47 @@ public struct CKBController: RouteCollection {
 
     // MARK: - API
 
-    func faucet(_ req: Request) -> Response {
-        let accessToken = req.http.cookies.all[accessTokenCookieName]?.string
-        let verifyStatus = Authorization().verify(accessToken: accessToken)
-        let result: [String: Any]
-
-        if verifyStatus == .tokenIsVailable {
-            let urlParameters = req.http.urlString.urlParametersDecode
-            do {
-                if let address = urlParameters["address"] {
-                    let txHash = try sendCapacity(address: address)
-                    Authorization().recordCollectionDate(accessToken: accessToken!)
-                    result = ["status": 0, "txHash": txHash]
-                } else {
-                     result = ["status": -3, "error": "No address"]
+    func faucet(_ req: Request) -> Future<Response> {
+        let urlParameters = req.http.urlString.urlParametersDecode
+        let accessToken = req.http.cookies.all[accessTokenCookieName]?.string ?? ""
+        let email = (try? GithubService.getUserInfo(for: accessToken).email) ?? ""
+        var isSucceed = false
+        var txHash = ""
+        return Authentication().verify(email: email, on: req).map { status -> String in
+            // Send capacity
+            if status == .tokenIsVailable {
+                do {
+                    if let address = urlParameters["address"] {
+                        txHash = try self.sendCapacity(address: address)
+                        isSucceed = true
+                        return ["status": 0, "txHash": txHash].toJson
+                    } else {
+                        return ["status": -3, "error": "No address"].toJson
```

Comment:
`-3`, `-4` and `0` are magic numbers. How about defining an enum of Int?

Reply:
https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/28e042d709ada331068461571cea010bb113037e